lib/router.js

```diff
@@ -0,0 +1,76 @@
+import { toChildArray, cloneElement } from 'preact'
+
+import { useEve } from './eve.js'
+import { frame } from './state.js'
+
+const { history, location } = window
+
+export const match = (route, path) => {
+  let props = {}
+  let key, val, i, j, v, k
+  key = val = ''
+  i = j = 0
+  while (i < route.length || j < path.length) {
+    v = path[i] || '/'
+    k = route[j] || '/'
+    if (k === '*') return (props[k] = path.slice(j)), props
+    if (val) {
+      if (v === '/' && k === '/') {
+        props[key] = val
+        val = ''
+        j++, i++
+      } else {
+        k !== '/' && ((key += k), j++)
+        v !== '/' && ((val += v), i++)
+      }
+    } else {
+      j++, i++
+      if (k === ':') (key = ''), (val = v)
+      else if (k !== v) return
+    }
+  }
+  return props
```

Comment:
This is the matching algorithm, it compares a route to a match, check the tests for details on how it's used and works.